### PR TITLE
Update choose method to use smallest util function

### PR DESCRIPTION
**Pull request by junchaowu**
@abhinav @blampe @breerly

### CODE REVIEW

File: tchannel/tornado/peer.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
 
 from collections import deque
 from itertools import takewhile, dropwhile
+
+from tchannel.container.heap import smallest, NoMatchError
```
Comment (abhinav) [inline tchannel/tornado/peer.py]: import order?

File: tchannel/tornado/peer.py [hunk 2]
```diff
@@ -735,19 +737,11 @@ def choose(self, hostport=None, blacklist=None):
...
-
-        return choose_peer
+        try:
+            return self.peer_heap.peek(
+                smallest(
+                    self.peer_heap, (lambda p: p.hostport not in blacklist),
+                ),
+            )
+        except NoMatchError:
+            return None
```
Comment (abhinav) [inline tchannel/tornado/peer.py]: Wouldn't you rather stick this onto the `PeerHeap` class?

Comment (abhinav) [inline tchannel/tornado/peer.py]: To clarify: Up until now, this class depended on only 'self.peer_heap` and its operations. Putting this logic here skips a level from this class' point of view. You're accessing `self.peer_heap` implementation details. You should add a `smallest` method to `PeerHeap` and call that from here.

Comment (junchaowu) [inline tchannel/tornado/peer.py]: good idea. I will move it to peerheap class.

**Comment by abhinav** [comment]
k

**Pull request merged**